internal/graph/check.go

```diff
@@ -311,7 +318,7 @@ func intersection(ctx context.Context, concurrencyLimit uint32, handlers ...Chec
 		select {
 		case result := <-resultChan:
 			if result.err != nil {
-				err = result.err
+				err = errors.Join(err, result.err)
 				continue
 			}
```

Comment:
Instead of checking for `ErrCycleDetected` twice below, can we instead check once here? 

```
case result := <-resultChan:
   if result.err == ErrCycleDetected {
      return &ResolveCheckResponse{
		Allowed: false,
		ResolutionMetadata: &ResolutionMetadata{
			DatastoreQueryCount: dbReads,
		},
	}, nil
   }
   if result.err != nil {
      err = errors.Join(err, result.err)
      continue
   }
```

Then much of the error handling below can be removed, right?

Reply:
Updated - does that look better?